core/unix/coredump.c

```diff
@@ -484,11 +484,17 @@ os_dump_core_internal(dcontext_t *dcontext)
     // of LOAD program header (number of sections minus one since shstrtab does not
     // require a LOAD header).
     const ELF_OFF program_header_count = section_count;
+    ELF_OFF file_offset = sizeof(ELF_HEADER_TYPE) +
+        sizeof(ELF_PROGRAM_HEADER_TYPE) * program_header_count +
+        PROGRAM_HEADER_NOTE_LENGTH;
+    const size_t padding = ALIGN_FORWARD(file_offset, os_page_size()) - file_offset;
```

Comment:
Add a comment explaining that even though the ELF spec doesn't say the section header table has to be page-aligned, in practice some tools seem to assume that it is.

Reply:
I've updated the comment to clarify it's the loadable segments.

Reply:
Wait, the "loadable segments" are not the "section headers": it's the latter, isn't it?  It is not the loadable segments: that's part of the program headers.